### ISSUE TITLE
Play-by-play Score State

### ISSUE THREAD

**Issue by adelage** [BurntSushi/nfldb#158]
Hi @BurntSushi,

Has any work been done to compile play-by-play score state? The goal here would be to use score state to qualify plays.

Some examples would be:
- "Give me all pass plays involving Tom Brady where the Patriots **are leading by at least 7 points**."
- "Give me all run plays by the Seahawks **where they are trailing** in the fourth quarter."

Thanks,
AD

**Comment by BurntSushi** [comment]
The only thing you can do is ask for the score at a particular point in the game: http://pdoc.burntsushi.net/nfldb#nfldb.Game.score_at_time

The queries you pose are fascinating, but would probably require specialized SQL to run efficiently (perhaps even a new table if the data needs to be rearranged).

**Comment by adelage** [comment]
Yeah, I can see from the source that you're essentially re-running history (for that game) every time **score_at_time** is run. Is there anything in the Gamecenter JSON related to score that could be added to the tables' base schema?

@BurntSushi

**Comment by BurntSushi** [comment]
Not to my knowledge. You'd have to build it yourself.

**Comment by adelage** [comment]
What are your thoughts on building it using a materialized view with a window function in the DB?

**Comment by BurntSushi** [comment]
I don't have any. I'd need to see a more detailed design.
